Домашка  спринт 13

IsaevMilan opened the pull request:
комит-1 крашится приложение при переходе на "Поиск"-исправлено
комит-2 Почему-то при нажатии на треки переходит на активити с медиатекой, а не плеером - ИСПРАВЛЕНО

IsaevMilan committed:
commit 1  Домашка  спринт 13
крашится приложение при переходе на "Поиск"

IsaevMilan committed:
commit 2  Домашка  спринт 13
1. крашится приложение при переходе на "Поиск"- исправлено
2. Почему-то при нажатии на треки переходит на активити с медиатекой, а не плеером

IsaevMilan committed:
commit 3  Домашка  спринт 13
Исправлены недочеты и ошибки
1. крашится приложение при переходе на "Поиск"- исправлено
2. Почему-то при нажатии на треки переходит на активити с медиатекой,а не плеером- исправлено

IsaevMilan committed:
commit 4  Домашка  спринт 13
Исправлены недочеты и ошибки
1. крашится приложение при переходе на "Поиск"- исправлено
2. Почему-то при нажатии на треки переходит на активити с медиатекой,а не плеером- исправлено
3. В истории остается только 1 трек- исправлено

faritowich commented on app/src/main/java/com/example/myplaylistmaker/MediaPlayer.kt:
```diff
@@ -0,0 +1,50 @@
+        album.text = intent.extras?.getString("Album") ?: "Unknown Album"
+        year.text = (intent.extras?.getString("Year") ?: "Year").take(4)
+        genre.text = intent.extras?.getString("Genre") ?: "Unknown Genre"
+        country.text = intent.extras?.getString("Country") ?: "Unknown Country"
```
Тут есть два основных момента, на которые рекомендую обратить внимание:
1) Сейчас строки для распознавания бандлов пишутся вручную тут и в MediaAdapter. То есть надо следить, чтобы не опечататься, плюс непонятно, из какого класса мы передаем эту экстру. Поэтому стоит выносить эти строки в константы в companion object и использовать их в классе, откуда передаем, и там, куда передаем.
2) Сейчас в это активити по кусочкам передаются примитивы сложного класса MediaData. На самом деле можно передавать сам класс MediaData. Достаточно сделать его реализующим интерфейс Parcelable и пометить аннотацией Parcelize (если не пометить, придется самому парсер писать). И получать в класс при помощи getParcelable. Рекомендую почитать про это документацию, обычно аргументы передаются именно так, а не по кусочкам, как сейчас.

IsaevMilan replied:
Хорошо буду пробовать)

faritowich commented on app/src/main/java/com/example/myplaylistmaker/MediaPlayer.kt:
```diff
@@ -0,0 +1,50 @@
+        year.text = (intent.extras?.getString("Year") ?: "Year").take(4)
+        genre.text = intent.extras?.getString("Genre") ?: "Unknown Genre"
+        country.text = intent.extras?.getString("Country") ?: "Unknown Country"
+        val getImage = (intent.extras?.getString("Cover") ?: "Unknown Cover").replace("100x100bb.jpg", "512x512bb.jpg")
```
Рекомендую вынести логику по преобразованию ссылки artworkUrl100 в другое разрешение в сам класс MediaData, создав там соответствующую переменную, например так:
```
    val artworkUrl512 
        get() = artworkUrl100.replace("100x100bb.jpg", "512x512bb.jpg")
```

faritowich commented on app/src/main/java/com/example/myplaylistmaker/MediaAdapter.kt:
```diff
@@ -1,31 +1,61 @@
+
+        App.getSharedPreferences()
+        holder.bind(media[position])
+        holder.itemView.setOnClickListener {
```
Можно также задавать клик лисенеры не в самом адаптере, а выносить их в активити, чтобы вынести из адаптера лишнюю логику, которая не отвечает непосредственно за показ айтемов. Вариантов реализации несколько, на эту тему можно почитать эту статью: https://antonioleiva.com/recyclerview-listener/

IsaevMilan replied:
ок

IsaevMilan committed:
commit 6  Домашка  спринт 13
Исправлены недочеты баги и ошибки
1. крашится приложение при переходе на "Поиск"- исправлено
2. Почему-то при нажатии на треки переходит на активити с медиатекой,а не плеером- исправлено
3. В истории остается только 1 трек- исправлено
4. Подкорректированы View элементы.
